contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -23,9 +24,16 @@ class GoCompile(GoWorkspaceTask):
   """
 
   @classmethod
+  @memoized_method
+  def implementation_version(cls):
```

Comment:
In theory, all it should take to accomplish this is to mark `--version` in `contrib/go/src/python/pants/contrib/go/subsystems/go_distribution.py` `fingerprint=True`...